sdk/storage/inc/blobs/internal/protocol/blob_rest_client.hpp

```
@@ -1002,6 +896,304 @@ namespace Azure { namespace Storage { namespace Blobs {
         return ListBlobContainersParseResponse(*response);
       }
 
+    private:
+      static ListContainersSegment ListContainersSegmentFromXml(XmlReader& reader)
```

If parsing XML does not except on partial XML, which sometimes gets returned by service due to unstable internet connection, then we should track if the content-length is being checked against the actual length of the response body within the Http Stack itself. Can you please create an item to track this?

---

From the rest client's point of view, the HTTP client returns a incomplete HTTP body. In this case HTTP client should throw an exception, right?

---

Yes, but CPPRestsdk does not, so we have this difficult to detect bug. It is due to a previous limitation of the Http stack that we should make sure don't exist in track 2.

---

I guess we can add a check, if request is not `HEAD` and length of HTTP response body doesn't match `Content-Length`, we throw an exception.

---

fixed

---

Does this checking work when a stream is used for response body?
BTW I don't think it should be here, I think we should ask Http stack to have this check.

---

removed